### PR TITLE
Updated UsesOwaspDependencyScan to read Gradle files

### LINKED ISSUE #272: The data provider for OWASP Dependency Check has to check gradle build
In reference to discussion https://github.com/SAP/fosstars-rating-core/pull/263#discussion_r453559229

The plugin usage and configuration check for gradle build projects are not in place.

This needs to be added:
- It should be in sync with `maven` check
- It should be handle both the features
  - `OWASP_DEPENDENCY_CHECK_USAGE`
  - `OWASP_DEPENDENCY_CHECK_FAIL_CVSS_THRESHOLD`

### PR DESCRIPTION
**Pull request by artem-smotrakov**
Here is a list of main updates:

- Updated the data provider to parse Gradle files and extract the feature values.
- Added new test cases.

This closes #272

### CODE REVIEW

File: src/main/java/com/sap/sgs/phosphor/fosstars/data/github/LocalRepository.java [hunk 1]
```diff
@@ -225,6 +225,23 @@ public void pull() throws IOException {
...
+   * @throws IOException If something went wrong.
+   */
+  public Optional<List<String>> readLinesOf(Path file) throws IOException {
+    Objects.requireNonNull(file, "On no! File name is null!");
+    Path path = info.path().resolve(file);
+    if (!Files.isRegularFile(path)) {
+      return Optional.empty();
+    }
+
+    return Optional.of(IOUtils.readLines(Files.newInputStream(path)));
```
Comment (sourabhsparkala) [inline src/main/java/com/sap/sgs/phosphor/fosstars/data/github/LocalRepository.java:242]: Please use `Files.newInputStream(path)` with `try-with-resources` block.

File: src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesOwaspDependencyScan.java [hunk 2]
```diff
@@ -111,13 +135,97 @@ private static void checkMaven(LocalRepository repository, ValueSet values) thro
...
+  private static void checkGradle(LocalRepository repository, ValueSet values) throws IOException {
+    for (Path gradleFile : repository.files(path -> path.getFileName().endsWith(".gradle"))) {
+      Optional<List<String>> something = repository.readLinesOf(gradleFile);
+      if (!something.isPresent()) {
+        return;
+      }
+      List<String> file = something.get();
+
+      if (foundOwaspDependencyCheckInGradle(file)) {
+        boolean isMainFile = "build.gradle".equals(gradleFile.toString().toString());
```
Comment (sourabhsparkala) [inline src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesOwaspDependencyScan.java:153]: Is toString().toString() necessary? Os is this a typo?

File: src/main/java/com/sap/sgs/phosphor/fosstars/data/github/LocalRepository.java (hunk 1 above)
Comment (artem-smotrakov) [reply src/main/java/com/sap/sgs/phosphor/fosstars/data/github/LocalRepository.java]: Good catch. It's now fixed. I opened #318 to make sure that we fixed all such issues.

File: src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesOwaspDependencyScan.java (hunk 2 above)
Comment (artem-smotrakov) [reply src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesOwaspDependencyScan.java]: It's a typo. It's not fixed.

**Comment by artem-smotrakov** [comment]
Thanks for the review @sourabhsparkala !

**Pull request merged**